managers/backup-manager/BackupStatusPoller.js

```diff
@@ -0,0 +1,277 @@
+                        }
+                      });
+                  } else {
+                    BackupStatusPoller.clearPoller(metadata.name, intervalId);
+                  }
+                })
+                .catch(AssertionError, err => {
+                  logger.error(`Error occured while polling for backup, marking backup as failed`, err);
+                  BackupStatusPoller.clearPoller(metadata.name, intervalId);
+                  return eventmesh.apiServerClient.updateResource({
+                    resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.BACKUP,
+                    resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+                    resourceId: metadata.name,
+                    status: {
+                      state: CONST.APISERVER.RESOURCE_STATE.FAILED,
+                      error: utils.buildErrorJson(err)
+                    }
+                  });
+                })
+                .catch(Conflict, () => {
+                  logger.debug(`Not able to acquire backup status poller processing lock for backup with guid ${object.metadata.name}, Request is probably picked by other worker`);
+                });
+            }
+          });
+        });
+    }
+
+    function startPoller(event) {
+      logger.debug('Received Backup Event: ', event);
+      return Promise.try(() => {
+        if ((event.type === CONST.API_SERVER.WATCH_EVENT.ADDED || event.type === CONST.API_SERVER.WATCH_EVENT.MODIFIED) && !BackupStatusPoller.pollers[event.object.metadata.name]) {
+          logger.debug('starting backup status poller for backup with guid ', event.object.metadata.name);
+          // Poller time should be little less than watch refresh interval as 
+          const intervalId = setInterval(() => poller(event.object, intervalId), config.backup.backup_restore_status_check_every);
+          BackupStatusPoller.pollers[event.object.metadata.name] = intervalId;
+        }
+      });
+    }
+    const queryString = `state in (${CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS},${CONST.APISERVER.RESOURCE_STATE.ABORTING})`;
+    return eventmesh.apiServerClient.registerWatcher(CONST.APISERVER.RESOURCE_GROUPS.BACKUP, CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP, startPoller, queryString)
+      .then(stream => {
+        logger.debug(`Successfully set watcher on backup resources for backup status polling with query string:`, queryString);
+        return Promise
+          .delay(CONST.APISERVER.POLLER_WATCHER_REFRESH_INTERVAL)
+          .then(() => {
+            logger.debug(`Refreshing backup stream after ${CONST.APISERVER.POLLER_WATCHER_REFRESH_INTERVAL}`);
+            stream.abort();
+            return this.start();
+          });
+      })
+      .catch(err => {
+        logger.error(`Error occured in registering watch for backup status poller:`, err);
+        return Promise
+          .delay(CONST.APISERVER.WATCHER_ERROR_DELAY)
```

Comment:
Is it better to read from config and default to constant value?

Reply:
I don't feel config is needed here.